lib/asciidoctor/parser.rb

```diff
@@ -1337,6 +1341,9 @@ def self.next_list_item(reader, list_block, match, sibling_trait = nil)
       # list
       while list_item_reader.has_more_lines?
         if (new_block = next_block(list_item_reader, list_item, {}, options))
+          if list_type == :dlist
+            list_item.source_location = list_item_reader.cursor if list_block.document.sourcemap
```

Comment:
This is not working anymore.
Given:

```
1 * level 1
2 ** level 2
3 *** level 3
4 **** level 4
5 ***** level 5
6 * level 1
```

When we are processing the first item of the nested list (ie. level 2), the cursor is already line 6.
Do we need to increment a counter in the while loop to keep track of the line number ?